### PR TITLE
Set up aeppl documentation

### PR DESCRIPTION
**Pull request by OriolAbril**
I had a call with @ricardoV94 (who can take over from here) to set up the documentation to match aeppl requirements and idea for the documentation. 

* Documentation is built with sphinx, and functions/modules/classes/methods are all automatically added to the documentation and have their docstring rendered. The only manual input needed is listing in `docs/source/api/index.md` the highest level functions on one side and modules on another, if one of those are added the file needs to be updated, if objects inside a module are added everything is updated automatically. This is achieved with autosummary and the two custom templates added in the PR.
* Documentation can be written with rst, markdown and jupyter notebook (both with the option of extra myst features to use all sphinx features from markdown). posts written as jupyter notebooks are executed when building the docs.
* readthedocs configuration is added too, along with the docs requirement files to keep doc building env minimal, only these requirements and aeppl are installed.
* A sample user guide post as a jupyter notebook is added.
* The github readme is included as documentation homepage
* the theme used is furo which keeps all the navigation tree on the left (no navbar) and has a page level table of contents on the right, this can be changed with two line in requirements and conf.py to for example sphinx-book-theme or other alternatives (though similar themes like readthedocs or alabaster don't have page table of contents on the right which helps following user guide pages)
* the makefile is updated to include commands `make html` and `make cleandocs`

This is still now working completely and needs some extra work which can be done here or in a follow up PR, what is done is working and can already be used.
* readthedocs needs to be activated on their site by someone with permissions
* there are modules that share a name with a top level function. We didn't know how to import those and therefore couldn't tell sphinx to import and document them. They are not present in the documentation.

Here are a few important guidelines and requirements to check before your PR can be merged:
+ [x] There is an informative high-level description of the changes.
+ [x] ~The description and/or commit message(s) references the relevant GitHub issue(s).~
+ [x] [`pre-commit`](https://pre-commit.com/#installation) is installed and [set up](https://pre-commit.com/#3-install-the-git-hook-scripts).
+ [x] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [x] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes), and there are **no commits that fix changes introduced by other commits in the same branch/BR**.  If your commit description starts with "Fix...", then you're probably making this mistake.
+ [x] ~There are tests covering the changes introduced in the PR.~

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
+.. |Tests Status| image:: https://github.com/aesara-devs/aeppl/actions/workflows/test.yml/badge.svg?branch=main
+    :target: https://github.com/aesara-devs/aeppl/actions/workflows/test.yml
+
+.. |Coverage| image:: https://codecov.io/gh/aesara-devs/aeppl/branch/main/graph/badge.svg?token=L2i59LsFc0
+    :target: https://codecov.io/gh/aesara-devs/aeppl
+
+.. |Gitter| image:: https://badges.gitter.im/aesara-devs/aeppl.svg
+   :alt: Join the chat at https://gitter.im/aesara-devs/aeppl
+   :target: https://gitter.im/aesara-devs/aeppl?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge
+
```
Comment (brandonwillard) [inline README.rst:15]: Why are these being moved?

**Review by brandonwillard — CHANGES_REQUESTED** [review]
Please provide a high-level description of the changes in the PR's opening comment.

File: README.rst (hunk 1 above)
Comment (OriolAbril) [reply README.rst]: we has some issues with the badge rendering and tried moving that to see if it helped, then accidentally committed the change after fixing the actual issue.

**Review by brandonwillard — COMMENTED** [review]
I've just set up RTD so that it's part of the PR CI process, so you view the documentation additions/changes after pushing to this PR.

Also, it looks like this branch is named `main`; that complicates use of the branch, because most people forking this repository will already have a `main`.  The feature branch naming convention that I recommend involves short hyphenated topic/task descriptions (e.g. `set-up-aeppl-docs`).

**Comment by OriolAbril** [comment]
> Also, it looks like this branch is named main;

I didn't even realize, sorry about that.

**Pull request merged**